### DISCUSSION TITLE
Is there a way to use this lib with ts automapper to be able to map this generated classes to dto classes?

### DISCUSSION THREAD

**Discussion by nfjdu** [omar-dulaimi/prisma-class-validator-generator#13] [Q&A]
So, i try to use [ts mapper](https://github.com/nartc/mapper) with this lib. Mapper requires to add some decorators to Model as well as to Dto.. So, how can we deal with all this things, knowing that each `generate` rewrites this models classes every time? Thanks for help

**Comment by omar-dulaimi** [comment]
Answering properly rather than leaving this open any longer. Sorry for the wait.

The constraint you hit is real and deliberate: everything under the output directory is regenerated wholesale on every `prisma generate`, so anything you add to a generated class is lost. That is what makes the output safe to trust, and it is also why decorating those classes directly cannot work.

**Keep your mapper decorators on a class you own**, and let the generated class stay generated:

```ts
import { UnitType } from './generated';        // generated, never edited
import { AutoMap } from '@automapper/classes';

export class UnitTypeDto {
  @AutoMap() id!: string;
  @AutoMap() length!: number;
}

createMap(mapper, UnitType, UnitTypeDto);
```

`createMap` needs metadata on both sides, so for the source you either use the `classes` strategy with `@AutoMap` on a hand-written mirror, or the `pojos` strategy, which takes a plain metadata description instead of decorators and is the better fit here precisely because it does not need to touch the generated file.

I am not going to pretend that is free: it is a second declaration of the shape, which is exactly the drift problem generators exist to remove. If your goal is one source of truth rather than mapper interop specifically, the sibling `prisma-zod-generator` now derives TypeScript types from generated Zod schemas rather than the other way round, so the type cannot drift from the validator. That is a different toolchain, not a drop-in answer to your question, but it is the direction that solves the underlying complaint.

Closing since the question has an answer. If you want the generator to emit a decorator hook of some kind, that is a feature request worth its own issue and I would read it.
